flake.nix

```diff
@@ -1,87 +1,40 @@
 {
   description = "Steel";
 
-  inputs = {
-    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable"; # We want to use packages from the binary cache
-    flake-utils.url = "github:numtide/flake-utils";
-  };
+  inputs.nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
 
   outputs = {
     self,
     nixpkgs,
-    flake-utils,
-  }:
-    flake-utils.lib.eachDefaultSystem (system: let
-      pkgs = nixpkgs.legacyPackages.${system};
-      fs = pkgs.lib.fileset;
+  }: let
+    systems = ["x86_64-linux" "aarch64-linux" "x86_64-darwin" "aarch64-darwin"];
+    eachSystem = nixpkgs.lib.genAttrs systems;
+    pkgsFor = nixpkgs.legacyPackages;
+  in {
+    packages = eachSystem (system: {
+      default = self.packages.${system}.steel;
+      steel = pkgsFor.${system}.callPackage ./nix/package.nix {
+        inherit (pkgsFor.${system}.darwin.apple_sdk.frameworks) Security;
+      };
+    });
+
+    formatter = eachSystem (system: pkgsFor.${system}.alejandra);
+    legacyPackages = self.packages;
+    defaultPackage = eachSystem (system: self.packages.${system}.default);
```

Comment:
Add a comment above these two that says `# DEPRECATED`. `legacyPackages` doesn't seem to need to be here, as the package set is flat, so it's already supported by `packages`.

`defaultPackage` is superseded by `packages.default`, which you already added.